src/js/controller/LayersListController.js

```diff
@@ -123,14 +124,50 @@
     }
   };
 
-  ns.LayersListController.prototype.renameCurrentLayer_ = function () {
+  ns.LayersListController.prototype.focusLayerElement_ = function(element) {
+    var index = pskl.utils.Dom.getData(element, 'layerIndex');
+    this.piskelController.setCurrentLayerIndex(parseInt(index, 10));
+
+    var selectedClass = 'current-layer-item';
+    this.currentLayerEl.classList.remove(selectedClass);
+    element.classList.add(selectedClass);
+    this.currentLayerEl = element;
+
+    this.updateButtonStatus_();
+  };
+
+  ns.LayersListController.prototype.onDoubleClick_ = function(evt) {
+    var el = evt.target || evt.srcElement;
+    if (el.classList.contains('layer-name')) {
+      this.openCurrentLayerRenameForm_();
+    }
+  };
+
+  ns.LayersListController.prototype.openCurrentLayerRenameForm_ = function() {
     var layer = this.piskelController.getCurrentLayer();
-    var name = window.prompt('Please enter the layer name', layer.getName());
-    if (name) {
+    this.currentLayerEl.classList.add('edit-name');
+    
+    var form = this.currentLayerEl.querySelector('.rename-layer-form');
+    pskl.utils.Event.addEventListener(form, 'submit', function(evt) {
+      evt.preventDefault();
+      var form = evt.target || evt.srcElement;
       var index = this.piskelController.getCurrentLayerIndex();
-      this.piskelController.renameLayerAt(index, name);
+      this.piskelController.renameLayerAt(index, form.layerName.value);
       this.renderLayerList_();
-    }
+      return false;
+    }, this);
+
+    var input = this.currentLayerEl.querySelector('.rename-layer-input-text');
+    input.value = layer.name;
+    pskl.utils.Event.addEventListener(input, 'blur', this.renderLayerList_, this);
+    pskl.utils.Event.addEventListener(input, 'keydown', function(evt) {
+      var eventKey = pskl.service.keyboard.KeyUtils.createKeyFromEvent(evt);
+
+      if (eventKey.key === 'ESC') {
```

Comment:
I felt the need to close form when ESC key is pressed, but key events are dismiss if fire inside an input.
I'm not cool with this solution, but I don't know if there's any.